include/IECoreScene/SceneInterface.h

```diff
@@ -283,6 +303,9 @@ class IECORESCENE_API SceneInterface : public IECore::RunTimeTyped
 		static CreatorMap &fileCreators();
 		static void registerCreator( const std::string &extension, IECore::IndexedIO::OpenMode modes, CreatorFn f );
 
+	private:
+		SceneInterfacePtr m_root;
```

Comment:
> SceneInterface is meant to be an abstract class, so this feels wrong.

Do you mean it's supposed to be an interface class with no storage and only pure virtual function signatures? If it did have this property what are the benefits?

Reply:
> Do you mean it's supposed to be an interface class with no storage and only pure virtual function signatures?

Yes.

> If it did have this property what are the benefits?

For me, trying to understand the code, one huge benefit would be clarity. I find the various SceneInterfaces are already a bit mind boggling, since they often have multiple internal implementation classes each with their own relationships. Pushing those relationships out into the base class really doesn't help.

Might this "all children have ownership of the root" setup also preclude certain implementations from storing their children for fast `child()` queries? Because it would force a circular reference? I have a nasty feeling I've had to avoid this pattern in the past for those reasons. Baking in assumptions about how these subclasses must be implemented seems wrong, especially because it's only to provide short term compatibility for tags. It's basically turning a hack into a structural feature.

Finally I guess it's the fact that we've been heading in a nice direction where we don't even expose the headers for our derived classes, so the implementation is all nicely hidden and changeable without breaking major version. This kindof breaks that.